pydantic_ai_slim/pydantic_ai/providers/github_models.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations as _annotations
+
+import os
+from typing import overload
+
+from httpx import AsyncClient as AsyncHTTPClient
+from openai import AsyncOpenAI
+
+from pydantic_ai.exceptions import UserError
+from pydantic_ai.models import cached_async_http_client
+from pydantic_ai.profiles import ModelProfile
+from pydantic_ai.profiles.cohere import cohere_model_profile
+from pydantic_ai.profiles.deepseek import deepseek_model_profile
+from pydantic_ai.profiles.grok import grok_model_profile
+from pydantic_ai.profiles.meta import meta_model_profile
+from pydantic_ai.profiles.mistral import mistral_model_profile
+from pydantic_ai.profiles.openai import OpenAIJsonSchemaTransformer, OpenAIModelProfile, openai_model_profile
+from pydantic_ai.providers import Provider
+
+try:
+    from openai import AsyncOpenAI
+except ImportError as _import_error:  # pragma: no cover
+    raise ImportError(
+        'Please install the `openai` package to use the GitHub Models provider, '
+        'you can use the `openai` optional group — `pip install "pydantic-ai-slim[openai]"`'
+    ) from _import_error
+
+
+class GitHubModelsProvider(Provider[AsyncOpenAI]):
+    """Provider for GitHub Models API.
+
+    GitHub Models provides access to various AI models through an OpenAI-compatible API.
+    See <https://docs.github.com/en/github-models> for more information.
+    """
+
+    @property
+    def name(self) -> str:
+        return 'github_models'
+
+    @property
+    def base_url(self) -> str:
+        return 'https://models.github.ai/inference'
+
+    @property
+    def client(self) -> AsyncOpenAI:
+        return self._client
+
+    def model_profile(self, model_name: str) -> ModelProfile | None:
+        model_name_lower = model_name.lower()
+
+        # GitHub Models uses prefixes like "xai/", "microsoft/", "meta/", etc.
+        prefix_to_profile = {
+            'xai/': grok_model_profile,
+            'meta/': meta_model_profile,
+            'microsoft/': openai_model_profile,
+            'mistral-ai/': mistral_model_profile,
+            'cohere/': cohere_model_profile,
+            'deepseek/': deepseek_model_profile,
+        }
+
+        for prefix, profile_func in prefix_to_profile.items():
+            if model_name_lower.startswith(prefix):
```

Comment:
Can we base this on `OpenRouterProvider.model_profile` instead, which splits on `/` instead of doing prefix matching?

Reply:
Oh nice, that's a better model for this code.